datafusion/core/src/physical_plan/planner.rs

```diff
@@ -174,9 +174,9 @@ fn create_physical_name(e: &Expr, is_first_expr: bool) -> Result<String> {
             let expr = create_physical_name(expr, false)?;
             Ok(format!("{} IS NOT UNKNOWN", expr))
         }
-        Expr::GetIndexedField { expr, key } => {
-            let expr = create_physical_name(expr, false)?;
-            Ok(format!("{}[{}]", expr, key))
+        Expr::GetIndexedField(get_indexed_field) => {
+            let expr = create_physical_name(&get_indexed_field.expr, false)?;
+            Ok(format!("{}[{}]", expr, get_indexed_field.key))
```

Comment:
I recommend this style so that if we ever change `GetIndexedField` the compiler will tell us all the places where a new field may be needed


```suggestion
        Expr::GetIndexedField(GetIndexedField { expr, key }) => {
            let expr = create_physical_name(expr, false)?;
            Ok(format!("{}[{}]", expr, key))
```